lib/faye-redis-sharded.js

```diff
@@ -289,20 +291,40 @@ Engine.prototype = {
         return self._ns + '/channels' + c;
       });
 
-    var notify = function (error, clients) {
-      if (error) return;
+    var notify = function (error, clientIds) {
+      if (error) {
+        return;
+      }
 
-      clients.forEach(function (clientId) {
-        var shard = self._getShard(clientId, shardManager),
-          redis = shard.redis;
+      const processNotification = (clientId) => {
+        const shard = self._getShard(clientId, shardManager);
+        const redis = shard.redis;
         self._server.debug('Queueing for client ?: ?', clientId, message);
-        redis.rpush(self._ns + '/clients/' + clientId + '/messages', jsonMessage, function (err, result) {
-          redis.publish(self._ns + '/' + clientId + '/notify', clientId);
+        redis.rpush(`${self._ns}/clients/${clientId}/messages}`, jsonMessage, function (err, result) {
+          redis.publish(`${self._ns}/${clientId}/notify`, clientId);
           self._server.debug('Published for client ? - ? - to server ?', clientId, message, shard.shardName);
         });
         self.clientExists(clientId, function(exists) {
-          if (!exists) redis.del(self._ns + '/clients/' + clientId + '/messages');
+          if (!exists) {
+            redis.del(`${self._ns}/clients/${clientId}/messages`);
+          }
         });
+      };
+
+      const start = new Date();
+      const clientIdsFiltered = [...new Set(clientIds)].filter(x => x); // remove nulls/duplicates
```

Comment:
Thanks for the comment (the `x => x` does what I assumed, but I wasn't certain...)